### PR TITLE
[pkg/golden] internal/coreinternal/golden -> pkg/golden

### LINKED ISSUE #28594: Move internal/golden to pkg/golden
### Component(s)

_No response_

### Is your feature request related to a problem? Please describe.

I have found the internal/golden package useful and would like to draw more on it. Unfortunately, I cannot import it in my projects as it is internal.

### Describe the solution you'd like

Move internal/golden to pkg/golden.

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Move the golden package to pkg/golden so it can be available to downstream projects.

**Link to tracking Issue:**
Fixes #28594

### CODE REVIEW

**Comment by atoulme** [comment]
Not sure why crosslink gives me a hard time. Any idea?

File: pkg/golden/go.mod [hunk 1]
```diff
@@ -0,0 +1,30 @@
+module golden
```
Comment (djaglowski) [inline pkg/golden/go.mod:1]:
```suggestion
module github.com/open-telemetry/opentelemetry-collector-contrib/pkggolden
```

Pretty sure this is causing a lot of the problems.

File: connector/exceptionsconnector/go.mod [hunk 2]
```diff
@@ -4,6 +4,7 @@ go 1.20
 
 require (
 	github.com/open-telemetry/opentelemetry-collector-contrib/internal/coreinternal v0.88.0
+	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden v0.0.0-00010101000000-000000000000
```
Comment (songy23) [inline connector/exceptionsconnector/go.mod:7]: https://github.com/open-telemetry/opentelemetry-collector-contrib/actions/runs/6650933955/job/18071954256?pr=28636#step:5:496

> go: github.com/open-telemetry/opentelemetry-collector-contrib/pkg/golden@v0.0.0-00010101000000-000000000000: invalid version: unknown revision 000000000000

**Comment by sakulali** [comment]
Last time i move kafka authentication config to internal module, it seems need to run commands below: 
make gotidy
make golint
make crosslink
make multimod-verify
make gendependabot
make genotelcontribcol
make docker-otelcontribcol

**Comment by atoulme** [comment]
Thanks for all the help!

**Comment by djaglowski** [comment]
Just need to run `make goporto`

**Review by djaglowski — APPROVED** [review]

**Pull request merged**